### ISSUE TITLE
[BUG] npm CLI has weird logs interrupting terminal usage in the presence of an `.npmrc` file.

### ISSUE THREAD

**Issue by tom-ricci** [npm/cli#4680]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

While running npm scripts, npm logs `timing config:load:flatten completed in 1/2ms` (The 1/2ms isnt what it actually logs, rather it flickers between 1ms and 2ms).

This only happens when I have a `.npmrc` file in the package root. npm also says `npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc` before starting my script.

While this does not prevent me from typing (as far as I can tell), I can't see my input nor is it really ideal for my terminal to do this. I also noticed a significant slowdown installing packages (part of my script has to run npm i -D "a bunch of packages"), although this could be a coincidence.

Example:
```shell
npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

> create-octobox-app@2.0.0 test
> npm run elevate && npm run build && npm run install && node test/main.test.js && npm run clean

npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

> create-octobox-app@2.0.0 elevate
> sudo echo '' || exit 0

[sudo] password for thomas: (⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ : timing config:load:flatten (⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ : timing config:load:flatten Completed in 2ms
(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ : timing config:load:flatten Completed in 1ms
npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

> create-octobox-app@2.0.0 build
> tsc --project tsconfig.json && minify bin/index.max.js > bin/index.js && rimraf bin/index.max.js && chmod u+x bin/index.js || exit 0

npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

> create-octobox-app@2.0.0 install
> npm run clean && sudo npm link || npm link

npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

> create-octobox-app@2.0.0 clean
> sudo npm -g uninstall create-octobox-app || npm -g uninstall create-octobox-app


up to date, audited 1 package in 147ms

found 0 vulnerabilities
npm WARN ignoring workspace config at /home/thomas/Documents/octobox/create-octobox-app/.npmrc 

added 1 package, and audited 3 packages in 497ms

found 0 vulnerabilities
➤ Running tests...⠂) ⠙ : timing config:load:flatten Completed in 2ms
➤ Running test 1...) ⠙ : timing config:load:flatten Completed in 2ms
```

### Expected Behavior

For this to not happen.

Same example as above, but what it should look like:
```shell
> create-octobox-app@2.0.0 test
> npm run elevate && npm run build && npm run install && node test/main.test.js && npm run clean


> create-octobox-app@2.0.0 elevate
> sudo echo '' || exit 0

[sudo] password for thomas: 


> create-octobox-app@2.0.0 build
> tsc --project tsconfig.json && minify bin/index.max.js > bin/index.js && rimraf bin/index.max.js && chmod u+x bin/index.js || exit 0


> create-octobox-app@2.0.0 install
> npm run clean && sudo npm link || npm link


> create-octobox-app@2.0.0 clean
> sudo npm -g uninstall create-octobox-app || npm -g uninstall create-octobox-app


removed 1 package, and audited 1 package in 157ms

found 0 vulnerabilities

added 1 package, and audited 3 packages in 349ms

found 0 vulnerabilities
➤ Running tests...
➤ Running test 1...

```

### Steps To Reproduce

1. Have a `.npmrc` file in some package installed locally with `npm link`. Mine had one line with `engine-strict=true` in it.
2. Run any of said package's npm scripts, anywhere on the system. For when I'm out of the package's root directory, I've used `npm create octobox-app`, because I have a binary to execute with `npx` in my package and the name is `create-octobox-app`. I haven't tested running a script without using `npx` or some sort of alias like `npm create` outside of the root directory, though.
3. See the error.

### Environment

- npm: 8.5.5
- Node.js: 16.14.0
- OS Name: Ubuntu 21.09
- System Model Name: Custom build (will provide specs if requested)
- npm config:
```ini
; "user" config from /home/thomas/.npmrc

//registry.npmjs.org/:_authToken = (protected) 
unsafe-perm = true 
user = "0" 

; node bin location = /usr/local/bin/node
; cwd = /home/thomas/Documents/octobox/create-octobox-app
; HOME = /home/thomas
; Run `npm config ls -l` to show all defaults.

```

**Comment by lukekarrys** [comment]
I believe this is a duplicate of https://github.com/npm/cli/issues/4527. I'm not going to close either one yet, since I'm not 100% sure but I'm linking them so I can investigate together.

**Comment by tom-ricci** [comment]
Aha, my issue does look like a duplicate of this. I would be able to just move my `.npmrc` up to my workspace root if it didn't need to be uploaded to the registry, but I'm using the file to force node and npm to be at least v8 and v16 on users machines, so I believe I need the `.npmrc` in the package's root so it will be pushed to the registry on publish. If I can do the same thing with it in the workspace, that would also work although I don't think I can do that, unless I'm missing something.

**Comment by lukekarrys** [comment]
What you are doing is perfectly valid to get the `.npmrc` file to the registry. As a side note, I think you need to explicitly opt-in to publishing this file (such as adding it to `package.json#files`).

The bug is (I think) that when a `warning` or `error` log message comes before the timing info then it sticks in the terminal when it shouldn't. Since I've confirmed this is the same as #4527, I'm going to close this one and track progress over there.
